core/storage/question/gae_models.py

```diff
@@ -354,15 +354,26 @@ def get_question_skill_links_based_on_difficulty_equidistributed_by_skill(
 
         question_skill_link_mapping = {}
 
+        def get_offset(query):
+            """Helper function to get the offset."""
+            question_count = query.count()
+            if question_count > 2 * question_count_per_skill:
+                return utils.get_random_int(
+                    question_count - (question_count_per_skill * 2))
+            return 0
+
         for skill_id in skill_ids:
-            query = cls.query(cls.skill_id == skill_id)
+            query = cls.gql("WHERE skill_id = '" + skill_id + "'") # pylint: disable=invalid-string-quote
```

Comment:
Sorry, what's the point of changing this?

Reply:
I am sorry I didn't realize that query also had an offset. Done.